src/main/java/frc/robot/commands/PositionRecorder.java

```diff
@@ -0,0 +1,119 @@
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands;
+
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import edu.wpi.first.networktables.EntryListenerFlags;
+import edu.wpi.first.networktables.EntryNotification;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.wpilibj.geometry.Pose2d;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.DriveTrain;
+
+public class PositionRecorder extends CommandBase {
+  /**
+   * Creates a new PositionRecorder.
+   */
+  public boolean isRunning = false;
+
+  DriveTrain drivetrain;
+  PrintWriter writer;
+  long start;
+
+  String directoryName="position-recordings";
+
+  NetworkTableEntry directoryNameEntry;
+  NetworkTableEntry isRunningEntry;
+  
+  public PositionRecorder(DriveTrain drivetrain) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    initSmartDashboard();
+    this.drivetrain = drivetrain;
+  }
+  public void initSmartDashboard(){
+    ShuffleboardTab tab =Shuffleboard.getTab("Position Recorder");
+    isRunningEntry=tab.add("Is running", isRunning).withWidget("Toggle Button").getEntry();
+    directoryNameEntry=tab.add("Directory Name", directoryName).getEntry();
+
+
+    isRunningEntry.addListener((EntryNotification e)-> setIsRunning(e.value.getBoolean()), EntryListenerFlags.kUpdate|EntryListenerFlags.kNew);
+
+    directoryNameEntry.addListener((EntryNotification e)-> directoryName=e.value.getString(), EntryListenerFlags.kUpdate|EntryListenerFlags.kNew);
+  }
+
+  public void setIsRunning(boolean running){
+    System.out.println("Set is running: "+running);
+    if(running!=isRunning){
+      if(running){
+        this.schedule();
+      }else{
+        this.cancel();
+      }
+    }
+    if(isRunningEntry.getBoolean(false)!=running){
+      isRunningEntry.setBoolean(running);
+    }
+  }
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    isRunning = true;
+    writer=null;
+    System.out.println("Position Recorder Begin");
+    try {
+      String filename = new SimpleDateFormat("MM-dd_HH_mm_ss").format(new Date())+".csv";
+
+      new File(directoryName).mkdirs();
+
+      File f = new File(directoryName,filename);
+      f.createNewFile();
+      writer = new PrintWriter(f); // PrintWriter is buffered
+      writer.println("Elapsed milliseconds,x,y,rotation");
+
+      start=System.currentTimeMillis();
+
```

Comment:
Why milliseconds? I think Robot.time() in seconds is more useful.

Reply:
I would prefer seconds. I think it's easier to read.